vkcs/backup/backup_shared.go

```diff
@@ -0,0 +1,273 @@
+package backup
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/clients"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/backup/v1/plans"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/backup/v1/providers"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/backup/v1/triggers"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/db/v1/instances"
+)
+
+const (
+	ProviderNameNova  = "cloud_servers"
+	ProviderNameTrove = "dbaase"
+	Nova              = "OS::Nova"
+	Trove             = "OS::Trove"
+	NovaInstance      = "OS::Nova::Server"
+	TroveInstance     = "OS::Trove::Instance"
+
+	RetentionFull = "max_backups"
+	RetentionGFS  = "gfs"
+
+	TimeFormat = "15:04"
+)
+
+var providerNameMapping = map[string]string{
+	"OS::Nova":  ProviderNameNova,
+	"OS::Trove": ProviderNameTrove,
+}
+
+func getResourcesInfo(config clients.Config, region string, instancesID []types.String, resourceType string) ([]*plans.BackupPlanResource, error) {
+	if resourceType == ProviderNameNova {
+		computeClient, err := config.ComputeV2Client(region)
```

Comment:
would be better to extract this to a separate method

Reply:
Extracted